### PR TITLE
Fix some retain cycles, improve code oragnization and style. Small refactor

### PR DESCRIPTION
**Pull request by jasonhotsauce**
@l5t @SpencerP Please take a look. I would like this to be merged then start further works. 

Thank you.

### CODE REVIEW

File: Wit/WITCircleView.m [hunk 1]
```diff
@@ -12,22 +12,18 @@ @interface WITCircleView ()
...
 
 #pragma mark - Lifecycle
-- (id)init {
+- (instancetype)init {
     self = [super init];
-    if (self) {
-        self.circleLayer = (WITCircleLayer*)self.layer;
+    if (!self) {
+        return nil;
+    }
```
Comment (SpencerP) [inline Wit/WITCircleView.m]: This is not a very common pattern in my experience. Can we stick to
```
self = [super init];
if (self) {
  // setup
}
```

Or, if you want to save a line:
```
if (self = [super init]) {
  // setup
}
```

File: Wit/RecordingSessionStarter.h [hunk 2]
```diff
@@ -1,17 +0,0 @@
...
-
-#import <Foundation/Foundation.h>
-#import "WITRecorder.h"
-
-@protocol RecordingSessionStarter <NSObject>
-
--(void)sessionDidStart:(WITRecorder *)recorder;
--(void)sessionEnded:(WITRecorder *)recorder;
-
-@end
```
Comment (SpencerP) [inline Wit/RecordingSessionStarter.h]: Why did we delete this file? Does this fit under "Fix some retain cycles" or  "improve code organization and style'?

File: Wit/WITMicButton.m [hunk 3]
```diff
@@ -277,10 +252,10 @@ - (void)newAudioLevel:(NSNotification*)n {
 - (id)initWithFrame:(CGRect)frame
 {
     self = [super initWithFrame:frame];
-    if (self) {
-        [self initialize];
-        self.frame = frame;
+    if (!self) {
+        return nil;
     }
```
Comment (SpencerP) [inline Wit/WITMicButton.m]: Same comment as above.

File: Wit/WITMicButton.m [hunk 4]
```diff
@@ -209,31 +209,6 @@ - (void)twoPulses {
...
-#pragma mark - KVO
-- (void)observeValueForKeyPath:(NSString *)keyPath
-                      ofObject:(id)object
-                        change:(NSDictionary *)change
-                       context:(void *)context {
-    if (object == self && [keyPath isEqualToString:@"frame"]) {
-        [self recomputePositions];
-        return;
-    }
-}
```
Comment (SpencerP) [inline Wit/WITMicButton.m]: Thanks for cleaning up this craziness.

**Comment by SpencerP** [comment]
Love the cleanup. Just a couple of nits and questions before I merge.

File: Wit/WITCircleView.m (hunk 1 above)
Comment (jasonhotsauce) [inline Wit/WITCircleView.m]: 👍 Will do tonight.

File: Wit/RecordingSessionStarter.h (hunk 2 above)
Comment (jasonhotsauce) [inline Wit/RecordingSessionStarter.h]: Because this file is never used so just clean it up.

File: Wit/WITMicButton.m (hunk 3 above)
Comment (jasonhotsauce) [inline Wit/WITMicButton.m]: 👍

**Comment by jasonhotsauce** [comment]
@SpencerP I made some changes based on the feedback. Can you please take a look?

**Pull request merged**